lib/Data/Database.php

```diff
@@ -691,21 +752,35 @@ private static function _getPrimaryKeyClauses($key = 'dataid')
      */
     private static function _getDataType()
     {
-        return self::$_type === 'pgsql' ? 'TEXT' : 'BLOB';
+        return self::$_type === 'pgsql' ? 'TEXT' : (self::$_type === 'oci' ? 'VARCHAR2(4000)' : 'BLOB');
```

Comment:
I guess converting that into a switch statement now it checks two different values would make the code more readable.